spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/pubsub/GcpPubSubAutoConfiguration.java

```diff
@@ -381,4 +383,12 @@ public SubscriptionAdminClient subscriptionAdminClient(
 	public TransportChannelProvider transportChannelProvider() {
 		return InstantiatingGrpcChannelProvider.newBuilder().build();
 	}
+
+	@Bean
+	@ConditionalOnMissingBean
+	@ConditionalOnClass(Flux.class)
```

Comment:
This must be moved to a separate inner `@Configuration` class, marked with this annotation.
Just on the `@Bean` method won't help us to avoid classpath issues.